integration-tests/hibernate-orm-panache/src/main/java/io/quarkus/it/panache/TestEndpoint.java

```diff
@@ -1337,10 +1337,18 @@ public String testProjection() {
         long countDistinct = projectionDistinctQuery.count();
         Assertions.assertEquals(1L, countDistinct);
 
+        PanacheQuery<Long> projectionDistinctQueryLong = Cat
+                // The spaces at the beginning are intentional
+                .find("   SELECT   disTINct  c.name as foo, cast(null as string) as bar, SUM(c.weight) as baz from Cat c where name = :name group by name  ",
+                        Parameters.with("name", bubulle.name))
+                .project(Long.class);
+        long countDistinctProjectLong = projectionDistinctQueryLong.count();
+        Assertions.assertEquals(1L, countDistinctProjectLong);
+
         // We are checking that not everything gets lowercased
         PanacheQuery<CatProjectionBean> letterCaseQuery = Cat
                 // The spaces at the beginning are intentional
-                .find("   SELECT   disTINct  'GARFIELD', 'JoN ArBuCkLe' from Cat c where name = :NamE group by name  ",
+                .find("   SELECT   disTINct  'GARFIELD' as foo, 'JoN ArBuCkLe' as bar from Cat c where name = :NamE group by name  ",
```

Comment:
This is suspicious, why this test suddenly requires a change?

Reply:
+1